sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2665,6 +2665,15 @@ object SQLConf {
       .checkValue(_ > 0, "The difference must be positive.")
       .createWithDefault(4)
 
+  val BROADCAST_HASH_JOIN_OUTPUT_PARTITIONING_EXPAND_LIMIT =
+    buildConf("spark.sql.execution.broadcastHashJoin.outputPartitioningExpandLimit")
+      .doc("The maximum number of partitionings that a HashPartitioning can be expanded to. " +
+        "This configuration is applicable only for inner joins.")
+      .version("3.1.0")
+      .intConf
+      .checkValue(_ > 0, "The value must be positive.")
```

Comment:
shall we allow 0 which disables this feature? can be useful when we want to benchmark the improvement of this feature.

Reply:
`1` should retain the same behavior (although it creates `PartitioningCollection` with the original `HashPartitioning`). I added an explicit check with `0` to disable this feature. Thanks.